Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php

```diff
@@ -21,236 +23,151 @@
-        // Postcode
-        $inputs['zipCode'] = $data['billing']['postcode'];
-        if ( ! $inputs['zipCode'] ) {
-            $inputs['zipCode'] = $data['shipping']['postcode'];
+
+        /* Extract the customer details. */
+        $customer = [ 'identifier' => (0 == $data['customer_id']) ? ('_0_' . rand()) : ('_' . $data['customer_id'])
+                    , 'firstName' => ($data['billing']['first_name']) ? ($data['billing']['first_name']) : ($data['shipping']['first_name'])
+                    , 'lastName' => ($data['billing']['last_name']) ? ($data['billing']['last_name']) : ($data['shipping']['last_name'])
+                    , 'country' => ($data['billing']['country']) ? ($data['billing']['country']) : ($data['shipping']['country'])
+                    /* , 'state' => ($data['billing']['state']) ? ($data['billing']['country']) : ($data['shipping']['country']) */
+                    , 'city' => ($data['billing']['city']) ? ($data['billing']['city']) : ($data['shipping']['city'])
+                    , 'address' => ($data['billing']['address_1']) ? ($data['billing']['address_1']/* . ' ' . $data['billing']['address_2']*/) : ($data['shipping']['address_1']/* . ' ' . $data['shipping']['address_2']*/)
+                    , 'zipCode' => ($data['billing']['postcode']) ? ($data['billing']['postcode']) : ($data['shipping']['postcode'])
+                    /* , 'phone' => $data['billing']['phone'] */
+                    , 'email' => $data['billing']['email']
+                    , 'tags' => []
+                    ];
+
+        /* Extract the items details. */
+        $items = array();
+        foreach ( $order->get_items() as $item ) {
+          $items[] = [ 'item' => $item['name']
+                    , 'units' =>  $item['quantity']
+                    , 'unitPrice' => number_format( number_format( ( float )$item['subtotal'], 2) / number_format( ( float )$item['quantity'], 2 ), 2 )
+                    /* , 'type' => '' */
+                    /* , 'code' => '' */
+                    /* , 'vatPercent' => '' */
+                    /* , 'itemDescription' => '' */
+                    ];
         }
-        
-        // Checksum
-        ksort($inputs);
-        $query = http_build_query($inputs);
-        $encoded = hash_hmac('sha512', $query, $privateKEY, true);
-        $checksum = '';
-        $checksum = base64_encode($encoded);
-        
+
+        /* Build the data object to be posted to Twispay. */
+        $orderData = [ 'siteId' => $siteID
+                    , 'customer' => $customer
+                    , 'order' => [ 'orderId' => (isset( $_GET['tw_reload'] ) && $_GET['tw_reload']) ? ($data['id'] . '_' . time()) : ($data['id'])
+                                  , 'type' => 'purchase'
+                                  , 'amount' => $data['total']
+                                  , 'currency' => $data['currency']
+                                  , 'items' => $items
+                                  , 'tags' => []

```

Comment:
We should comment this line as `tags` is not a required field.

Reply:
Done